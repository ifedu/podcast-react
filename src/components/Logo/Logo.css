.Logo div {
  color: #7a98a1;
  font-weight: bold;
  margin-bottom: 8px;
}
