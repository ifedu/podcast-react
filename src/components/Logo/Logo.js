import React from 'react';
import { Link } from 'react-router-dom';

import './Logo.css';

export function Logo() {
  return (
    <Link to={'/'} data-testid='logo'>
      <div className='Logo'>
        <div>Podcaster</div>
      </div>
    </Link>
  );
}
