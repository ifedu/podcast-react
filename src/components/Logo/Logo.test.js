import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer' ;
import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { Logo } from './Logo';

const getContent = (pathname = '/') =>
  <MemoryRouter initialEntries={[{ pathname }]}>
    <div className='head'>
      <Logo/>
    </div>

    <Routes>
      <Route path='/' element={<div data-testid='main'></div>}></Route>
      <Route path='/foo' element={<div></div>}></Route>
    </Routes>
  </MemoryRouter>;

describe('Logo', () => {
  const content = getContent();

  afterEach(cleanup);

  it('renders correctly', () => {
    const tree = create(content).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the logo', () => {
    render(content);
    const logo = screen.getByTestId('logo');
    expect(logo).toBeInTheDocument();
  });

  it('navigates to the main page', async () => {
    const content = getContent('/foo');

    render(content);

    await waitFor(() => {
      const main = screen.queryByTestId('main');
      expect(main).toBeNull();
    });

    const logo = screen.queryByTestId('logo');
    userEvent.click(logo);

    await waitFor(() => {
      const main = screen.queryByTestId('main');
      expect(main).toBeInTheDocument();
    });
  });
});
