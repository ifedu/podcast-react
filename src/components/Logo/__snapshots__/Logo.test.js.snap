// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
Array [
  <div
    className="head"
  >
    <a
      data-testid="logo"
      href="/"
      onClick={[Function]}
    >
      <div
        className="Logo"
      >
        <div>
          Podcaster
        </div>
      </div>
    </a>
  </div>,
  <div
    data-testid="main"
  />,
]
`;
